src/state/transactions/updater.tsx

```diff
@@ -54,6 +57,24 @@ export default function Updater() {
         })
       )
 
+      const ttsMarks = performance.getEntriesByName('tts-mark', 'mark')
+      let elapsedTime: number | undefined
+      // Only log this metric for the first swap of a session.
+      if (ttsMarks.length === 0) {
+        elapsedTime = performance.now() / 1000 // time-to-swap in seconds
+        performance.mark('tts-mark')
+      }
+      if (elapsedTime && elapsedTime > 60 * 5) {
+        // 5 minute maximum
```

Comment:
think this would all be cleaner in a helper function that just returns `undefined` if there is > 1 entry 

and a bit confused on the 5 min metric here - is this meant to filter out swaps that just take foreverrr to get completed? i feel like we should still log those and just let data filter out according to whatever threshold they want?

Reply:
true, we can just filter on the backend!